### ISSUE TITLE
[mtt] UCX-CUDA: Segmentation fault: address not mapped to object at address (nil) 

### ISSUE THREAD

**Issue by avildema** [openucx/ucx#5240]
Configuration
```
OMPI: 4.0.4rc2
MOFED: MLNX_OFED_LINUX-5.0-1.0.0.0
Module: hpcx-gcc (2020-06-03)
Test module: none
Nodes: vulcan x2 (ppn=24(x2), nodelist=vulcan[02,05])
```

Comment
```
Cannot reproduce , but we see this error in mtt testing very often.
```
 
Cmd:
`/hpc/local/benchmarks/daily/next/2020-06-03/hpcx-gcc-redhat7.6/ompi/bin/mpirun -np 2 --display-map -mca btl self --tag-output --timestamp-output -mca pml ucx -mca coll '^hcoll' -x UCX_ERROR_SIGNALS=SIGILL,SIGSEGV,SIGBUS,SIGFPE,SIGABRT -x UCX_HANDLE_ERRORS=bt -x UCX_IB_PCI_RELAXED_ORDERING=off --bind-to core -x UCX_NET_DEVICES=mlx5_0:1 -mca osc '^ucx' -x UCX_IB_REG_METHODS=rcache,direct -x UCX_UD_TIMEOUT=10m -mca pmix_base_async_modex 1 -mca mpi_add_procs_cutoff 0 -mca pmix_base_collect_data 0 -x UCX_IB_NUM_PATHS=4 -x UCX_MAX_EAGER_LANES=4 -x UCX_MAX_RNDV_LANES=4 --map-by node /hpc/local/benchmarks/daily/next/2020-06-03/hpcx-gcc-redhat7.6/ompi/tests/osu-micro-benchmarks-5.6.2-cuda/get_local_rank /hpc/local/benchmarks/daily/next/2020-06-03/hpcx-gcc-redhat7.6/ompi/tests/osu-micro-benchmarks-5.6.2-cuda/osu_fop_latency`
 
Output:
```sh
Wed Jun  3 13:07:43 2020[1,0]<stderr>:[vulcan02:27906:0:27906] Caught signal 11 (Segmentation fault: address not mapped to object at address (nil))
Wed Jun  3 13:07:43 2020[1,0]<stderr>:==== backtrace (tid:  27906) ====
Wed Jun  3 13:07:43 2020[1,0]<stderr>: 0 0x0000000000051a45 ucs_debug_print_backtrace()  /build-result/src/hpcx-gcc-redhat7.6/ucx-master/src/ucs/debug/debug.c:656
Wed Jun  3 13:07:43 2020[1,0]<stderr>:=================================
Wed Jun  3 13:07:43 2020[1,0]<stderr>:[vulcan02:27906] *** Process received signal ***
Wed Jun  3 13:07:43 2020[1,0]<stderr>:[vulcan02:27906] Signal: Segmentation fault (11)
Wed Jun  3 13:07:43 2020[1,0]<stderr>:[vulcan02:27906] Signal code:  (-6)
Wed Jun  3 13:07:43 2020[1,0]<stderr>:[vulcan02:27906] Failing at address: 0x8cd00006d02
Wed Jun  3 13:07:43 2020[1,0]<stderr>:[vulcan02:27906] [ 0] Wed Jun  3 13:07:43 2020[1,0]<stderr>:/lib64/libpthread.so.0(+0xf5d0)[0x7fe94b4b15d0]
Wed Jun  3 13:07:43 2020[1,0]<stderr>:[vulcan02:27906] *** End of error message ***Wed Jun  3 13:07:44 2020[1,1]<stderr>:[vulcan05:12338:0:12338] Caught signal 11 (Segmentation fault: address not mapped to object at address (nil))
Wed Jun  3 13:07:44 2020[1,1]<stderr>:==== backtrace (tid:  12338) ====
Wed Jun  3 13:07:44 2020[1,1]<stderr>: 0 0x0000000000051a45 ucs_debug_print_backtrace()  /build-result/src/hpcx-gcc-redhat7.6/ucx-master/src/ucs/debug/debug.c:656
Wed Jun  3 13:07:44 2020[1,1]<stderr>:=================================Wed Jun  3 13:07:44 2020[1,1]<stderr>:[vulcan05:12338] *** Process received signal ***
Wed Jun  3 13:07:44 2020[1,1]<stderr>:[vulcan05:12338] Signal: Segmentation fault (11)
Wed Jun  3 13:07:44 2020[1,1]<stderr>:[vulcan05:12338] Signal code:  (-6)
Wed Jun  3 13:07:44 2020[1,1]<stderr>:[vulcan05:12338] Failing at address: 0x8cd00003032
Wed Jun  3 13:07:44 2020[1,1]<stderr>:[vulcan05:12338] [ 0] /lib64/libpthread.so.0(+0xf5d0)[0x7f4138c425d0]
Wed Jun  3 13:07:44 2020[1,1]<stderr>:[vulcan05:12338] *** End of error message ***--------------------------------------------------------------------------
Primary job  terminated normally, but 1 process returned
a non-zero exit code. Per user-direction, the job has been aborted.
----------------------------------------------------------------------------------------------------------------------------------------------------
mpirun noticed that process rank 0 with PID 0 on node vulcan02 exited on signal 11 (Segmentation fault).
```

**Comment by brminich** [comment]
we do not support all atomic cases on cuda yet

**Comment by avildema** [comment]
I see a lot a same problem in mtt testing with different tests.
Falls have common parameters such as: `-x UCX_IB_NUM_PATHS=4 -x UCX_MAX_EAGER_LANES=4 -x UCX_MAX_RNDV_LANES=4 ` and `-mca osc ^ucx`

**Comment by brminich** [comment]
can you reproduce some fail manually? If yes, pls add the details here

**Comment by avildema** [comment]
Yes, I can

`sh
salloc -p vulcan -N 2 -t 360
module load hpcx-gcc
module load hpcx-env/cuda
module load hpcx-env/gdrcopy

/hpc/local/benchmarks/daily/next/2020-06-07/hpcx-gcc-redhat7.6/ompi/bin/mpirun -np 2 -mca btl self --tag-output --timestamp-output -mca pml ucx -mca coll '^hcoll' -x UCX_ERROR_SIGNALS=SIGILL,SIGSEGV,SIGBUS,SIGFPE,SIGABRT -x UCX_HANDLE_ERRORS=bt -x UCX_IB_PCI_RELAXED_ORDERING=off --bind-to core -x UCX_NET_DEVICES=mlx5_0:1 -mca osc '^ucx' -x UCX_IB_REG_METHODS=rcache,direct -x UCX_UD_TIMEOUT=10m -mca pmix_base_async_modex 0 -mca mpi_add_procs_cutoff 100000 -mca pmix_base_collect_data 1 -x UCX_IB_NUM_PATHS=4 -x UCX_MAX_EAGER_LANES=4 -x UCX_MAX_RNDV_LANES=4 --map-by node /hpc/local/benchmarks/daily/next/2020-06-07/hpcx-gcc-redhat7.6/ompi/tests/osu-micro-benchmarks-5.6.2-cuda/get_local_rank /hpc/local/benchmarks/daily/next/2020-06-07/hpcx-gcc-redhat7.6/ompi/tests/osu-micro-benchmarks-5.6.2-cuda/osu_mbw_mr -W 16
`

**Comment by yosefe** [comment]
fixed by #5437 and #5442
